woocommerce/woocommerce-ios
# Refactor OrderStatusListViewController to move VM code into OrderDetailsViewModel to support VM refreshing Notes on Status changes

Fixes #3447 

To test:

- open any order
- change its status (all status changes result in an order note)
- after applying the status change, within a second or two, you should also see the order notes update with the status change note
- repeat the above steps - this time watch for a "undo" notice and tap on it to undo the change (hurry - it only appears briefly)
- ensure the change is undone and both changes appear in the order notes

Rationale for changes:

- I moved the data source/view model out of `OrderStatusListViewController` and into two new objects: `OrderStatusListViewModel` and `OrderStatusListDataSource`. I did this to reduce the concerns (and size) of OSLVC and to mirror what is done elsewhere in the app (e.g. product reviews)
- I changed `OrderStatusListViewController` to return, via closure, any requested change in status to `OrderDetailsViewController` instead of allowing it to initiate the action to change the status (and update notes). This way `OrderDetailsViewController` could decide what needs to happen in addition to the new status (e.g. re-fetching notes or showing or hiding other UI bits like shipping labels) and since the undo actions and notices that result from a status change are owned by ODVC
- I moved the action creation from `OrderStatusListViewController` to `OrderDetailsViewController` for this same reason
- I observed that, even without these changes, the pull-to-refresh behavior on `develop` was not working: #3608 #3609 - I removed the pull-to-refresh and will fix that in a follow on PR to help keep this already large PR from getting larger 

Update release notes:
